### PR TITLE
v1.4: Refactor stockup command to use stockup model interface

### PR DESCRIPTION
**Pull request by rebstan97**
`Stockup` command was implemented before the stockup model interface was introduced. Now the command is re-implemented to make use of the model interface.

### CODE REVIEW

**Comment by AZhiKai** [comment]
Just curious but was the command removed at some point?

**Comment by rebstan97** [comment]
> Just curious but was the command removed at some point?

Nope, the command was not removed. It was using the `updateIngredient()` method of the model interface, instead of the `stockup()` method

**Comment by AZhiKai** [comment]
> > Just curious but was the command removed at some point?
> 
> Nope, the command was not removed. It was using the `updateIngredient()` method of the model interface, instead of the `stockup()` method

Sounds good. :)

File: src/main/java/seedu/restaurant/logic/commands/ingredient/StockUpCommand.java [hunk 1]
```diff
@@ -35,22 +35,22 @@
...
-    private final Map<IngredientName, Integer> hashMap;
+    private final Map<IngredientName, Integer> linkedHashMap;
 
     /**
      * Creates a StockUpCommand to stock up the specified {@code Ingredient}
      */
     public StockUpCommand(Map<IngredientName, Integer> ingredientHashMap) {
         requireNonNull(ingredientHashMap);
-        this.hashMap = ingredientHashMap;
+        this.linkedHashMap = ingredientHashMap;
```
Comment (AZhiKai) [inline src/main/java/seedu/restaurant/logic/commands/ingredient/StockUpCommand.java]: Any reason for changing to LinkedHashMap? :o

Comment (rebstan97) [reply src/main/java/seedu/restaurant/logic/commands/ingredient/StockUpCommand.java]: The order isn't really important for the actual program functionality, but for the tests in asserting the correct success message

**Comment by AZhiKai** [comment]
Is the PR ready for review? 😮

File: src/test/java/seedu/restaurant/logic/commands/ingredient/StockUpCommandTest.java [hunk 2]
```diff
@@ -88,13 +78,11 @@ public void execute_multipleValidIngredients_success() {
...
-        ChangeStockDescriptor descriptor =
-                descriptorBuilder.buildChangeStockDescriptor("Iceberg Lettuce", 35);
-        descriptorList.add(descriptor);
+        Map<IngredientName, Integer> ingredientHashMap = new LinkedHashMap<>();
+        ingredientHashMap.put(new IngredientName("Iceberg Lettuce"), 10);
 
-        StockUpCommand stockUpCommand = new StockUpCommand(descriptorList);
-        String expectedMessage = String.format(Messages.MESSAGE_INGREDIENT_NAME_NOT_FOUND);
+        StockUpCommand stockUpCommand = new StockUpCommand(ingredientHashMap);
+        String expectedMessage = String.format(Messages.MESSAGE_STOCKUP_INGREDIENT_NOT_FOUND);
```
Comment (HyperionNKJ) [inline src/test/java/seedu/restaurant/logic/commands/ingredient/StockUpCommandTest.java]: I think this is a redundant call to String.format?

Comment (AZhiKai) [reply src/test/java/seedu/restaurant/logic/commands/ingredient/StockUpCommandTest.java]: Agree, it should be removed.

**Comment by rebstan97** [comment]
> Is the PR ready for review? 😮

Need to change the LinkedHashMap back to HashMap but otherwise yup!

File: src/test/java/seedu/restaurant/logic/commands/ingredient/StockUpCommandTest.java (hunk 2 above)
Comment (rebstan97) [reply src/test/java/seedu/restaurant/logic/commands/ingredient/StockUpCommandTest.java]: Good point! Will remove it

**Comment by AZhiKai** [comment]
> > Is the PR ready for review? 😮
> 
> Need to change the LinkedHashMap back to HashMap but otherwise yup!

Okay let us know when it's ready for review. Have removed the reviewers for now.

**Pull request merged**